streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -275,10 +275,6 @@ public void close(final Map<TopicPartition, Long> ackedOffsets) throws Processor
                     log.error("Failed to close state store {}: ", store.name(), e);
                 }
             }
-
-            if (ackedOffsets != null) {
-                checkpoint(ackedOffsets);
```

Comment:
One note, with the removal of this line, the `ackedOffsets` is no longer used at all in this method.  However, it's part of the `StateManager#close` interface, but it's not safe to remove as `GlobalStateManager` also implements the `StateManager` interface.  Which to me, brings up two questions 

1.  Do we need to apply the same approach for global state stores?
2. Should we refactor the `StateManager` interface to have a no-args `close()` method?

Given the time constraints, I don't think should hold up this PR but it's something that IMHO we shouldn't let slip.